smac_planner/src/a_star.cpp

```diff
@@ -348,7 +348,7 @@ bool AStarAlgorithm<NodeT>::createPath(
 template<typename NodeT>
 bool AStarAlgorithm<NodeT>::isGoal(NodePtr & node)
 {
-  return node == getGoal();
+  return *node == *getGoal();
```

Comment:
I really don't think this is necessary. It was working fine beforehand. If you really want to do this, we can do it as `node->getIndex()` instead, since that's what `operator==` for them are checking for.

Reply:
Offending commit removed.